evm/src/modules/TokenGateway.sol

```diff
@@ -41,58 +77,140 @@ contract TokenGateway is IIsmpModule {
         _;
     }
 
-    constructor(address _admin) {
+    constructor(address _admin, address _uniswapV2Router) {
         admin = _admin;
+        uniswapV2Router = IUniswapV2Router(_uniswapV2Router);
     }
 
     // set the ismp host address
     function setIsmpHost(address _host) public onlyAdmin {
         host = _host;
-        admin = address(0);
+    }
+
+    function setTokenIdentifiersERC20(bytes32 _tokenId, address _erc20) external onlyAdmin {
+        _erc20s[_tokenId] = _erc20;
+    }
+
+    function setTokenIdentifiersERC6160(bytes32 _tokenId, address _erc6160) external onlyAdmin {
+        _erc6160s[_tokenId] = _erc6160;
+    }
+
+    function setForeignTokenIdToLocalTokenId(bytes32 _foreignTokenId, bytes32 _localTokenId) external onlyAdmin {
+        _assets[_foreignTokenId] = _localTokenId;
+    }
+
+    function setChainsGateway(bytes memory _chain, address _host) external onlyAdmin {
+        _chainToGateway[_chain] = abi.encodePacked(_host);
```

Comment:
Can only be set once by admin. But can be updated through cross-chain governance